### PR TITLE
Use TaskGroup

### PR DESCRIPTION
**Pull request by swtwsk**
## Allow usage of `TaskGroup` in Airflow 2

Also, fixes tests incompatible with Airflow 1

---
Keep in mind:
- [ ] Documentation updates
- [x] [Changelog](CHANGELOG.md) updates

### CODE REVIEW

File: dbt_airflow_manifest_parser/builder.py [hunk 1]
```diff
@@ -27,21 +33,26 @@ def _make_dbt_run_task(self, model_name: str) -> BaseOperator:
...
                 model_name = node_name.split(".")[-1]
-                run_task = self._make_dbt_run_task(model_name)
-                test_task = self._make_dbt_test_task(model_name)
-                # noinspection PyStatementEffect
-                run_task >> test_task
-                tasks[node_name] = ModelExecutionTask(run_task, test_task)
+
+                is_first_version = airflow.__version__.startswith("1.")
+                task_group = None if (is_first_version or not use_task_group) else TaskGroup(group_id=model_name)
+                task_group_ctx = task_group or contextlib.nullcontext()
```
Comment (p-pekala) [inline dbt_airflow_manifest_parser/builder.py:45]: could you extract these 3 lines into separated method called eg: create_task_group

File: tests/test_dag_factory.py [hunk 2]
```diff
@@ -27,3 +28,61 @@ def test_dag_factory():
...
+    assert dag.default_args == {
+        "owner": "Piotr Pekala",
+        "email": ["test@getindata.com"],
+        "depends_on_past": False,
+        "start_date": datetime(2021, 10, 20, 0, 0, 0, tzinfo=timezone.utc),
+        "email_on_failure": False,
+        "email_on_retry": False,
+        "retries": 0,
+        "retry_delay": 300,
+    }
```
Comment (p-pekala) [inline tests/test_dag_factory.py:86]: if you test task groups you don't need to test again other parameters like descirption, schedule_iterval etc

File: dbt_airflow_manifest_parser/builder.py (hunk 1 above)
Comment (swtwsk) [reply dbt_airflow_manifest_parser/builder.py]: done

File: tests/test_dag_factory.py (hunk 2 above)
Comment (swtwsk) [reply tests/test_dag_factory.py]: you're right, I've deleted this part

**Review by p-pekala — APPROVED** [review]

**Pull request merged**